### PR TITLE
add lint precommit hook

### PR DESCRIPTION
**Pull request by nhanphan**

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+set +e
+
+# Stash un-staged changes - optional, see note below
```
Comment (danenbm) [inline .husky/pre-commit:5]: Nit: I don't see any note below for these two comments, but I see why we would stash, lint, then un-stash

File: .husky/pre-commit
Comment (danenbm) [inline .husky/pre-commit:1]: So I don't know how this is connected to git but I guess just having it in a file called pre-commit makes it run.  The `prepare` script in package.json doesn't need to be explicitly called I guess.

**Review by danenbm — APPROVED** [review]
Very cool thank you for adding!

File: .husky/pre-commit
Comment (nhanphan) [reply .husky/pre-commit:1]: the npm will automatically run prepare after install if it is specified in the package.json

**Pull request merged**